agents/cryomech-cpa1114/cryomech_cpa1114_agent.py

```diff
@@ -78,24 +78,33 @@ def breakdownReplyData(self, rawdata):
                   "High Pressure": [46,45,48,47], "High Pressure Average": [50, 49, 52, 51], "Delta Pressure": [54, 53, 56, 55],
                   "Motor Current": [58, 57, 60, 59]}
         
-        #Iterate through all keys and return the data in a usable format
+        #Iterate through all keys and return the data in a usable format. If there is an error in the string format, print the  
+        # error to logs, return an empty dictionary, and flag the data as bad
         data = {}
-        for key in keyloc.keys():
-            locs = keyloc[key] 
-            wkrBytes = bytes([rawdata[loc] for loc in locs])
-            
-            #three different data formats to unpack
-            if key in ["Operating State", "Pump State"]:
-                state = int.from_bytes(wkrBytes, byteorder='big')
-                data[key] = state
-            
-            if key in ["Warnings", "Alarms"]:
-                data[key] = int(''.join('{:02x}'.format(x) for x in wkrBytes))
-                
-            if key in ["Coolant In", "Coolant Out", "Oil", "Helium", "Low Pressure", "Low Pressure Average","High Pressure", "High Pressure Average", "Delta Pressure","Motor Current"]:
-                data[key] = struct.unpack('f', wkrBytes)[0]
-                
-        return data
+        
+        try:
+            for key in keyloc.keys():
+                locs = keyloc[key] 
+                wkrBytes = bytes([rawdata[loc] for loc in locs])
+
+                #three different data formats to unpack
+                if key in ["Operating State", "Pump State"]:
+                    state = int.from_bytes(wkrBytes, byteorder='big')
+                    data[key] = state
+
+                if key in ["Warnings", "Alarms"]:
+                    data[key] = int(''.join('{:02x}'.format(x) for x in wkrBytes))
```

Comment:
You need to pass a radix of 16 to the `int` cast, so it decodes the string as hexadecimal.  Currently this will either crash (if the string contains a-f) or decode incorrectly (because it's interpreting digits as decimal).  Just add a second argument `int(..., 16)`.

Reply:
Ah, this is probably the source of the crashes we have been seeing!  When there is a warning, the string is something like this: 'c9000000', which currently the system can't handle. Cryomech directly gave us this code, so I'll forward this correction to them after verifying this is really the solution.